src/prefect/server/models/artifacts.py

```diff
@@ -48,3 +50,127 @@ async def read_artifact(
 
     result = await session.execute(query)
     return result.scalar()
+
+
+@inject_db
+async def _apply_artifact_filters(
```

Comment:
Hmmm this doesn't quite look right to me.

Reply:
Can we also add a test that uses filtering?

Reply:
Yeah, it wasn't right, good catch!

Reply:
I'll need add this into the `nebula` as well